File: q201.py
class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        if left==0 or right==0:
            return 0;
        l_msb=int(math.log(left,2))
        r_msb=int(math.log(right,2))
        res=0
        while(l_msb==r_msb):
            res+=1<<l_msb
            left-=1<<l_msb
            right-=1<<l_msb
            if(left<=0 or right<=0):
                return res
            l_msb=int(math.log(left,2))
            r_msb=int(math.log(right,2))
        return res

File: q201.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeBitwiseAnd = function(left, right) {
    if(left==0 || right==0) return 0;
    if(left==right) return left;
    let l_msb=parseInt(Math.log(left)/Math.log(2));
    let r_msb=parseInt(Math.log(right)/Math.log(2));
    let res=0;
    while(l_msb==r_msb){
        res+=1<<l_msb;
        left-=1<<l_msb;
        right-=1<<l_msb;
        if(left<=0||right<=0) return res;
        l_msb=parseInt(Math.log(left)/Math.log(2));
        r_msb=parseInt(Math.log(right)/Math.log(2));
    }
    return res;
};
